### PR TITLE
chore: push metrics and code cleanup

### PR DESCRIPTION
**Pull request by ami-aman**
Linear ticket: https://linear.app/customerio/issue/MBL-545/push-metrics-tracking-ios-and-android-rn

### Problem:
Our package provides push metrics as a feature. We want to provide same feature using CDP native modules.

### Changes:
This PR includes:
- Code cleanup
- Fixed and removed TODOs wherever possible

TODO: Change base branch to `feature/cdp-main`

### CODE REVIEW

File: android/src/main/java/io/customer/reactnative/sdk/messagingpush/RNCIOPushMessaging.kt [hunk 1]
```diff
@@ -32,6 +34,10 @@ class RNCIOPushMessaging(
      */
     private var notificationRequestPromise: Promise? = null
 
+    private val SDKComponent.pushModuleConfig: MessagingPushModuleConfig
+        get() = newInstance {
+            modules["MessagingPushFCM"]?.moduleConfig as MessagingPushModuleConfig
```
[the remark this answers is not in the document]
Comment (mrehan27) [reply android/src/main/java/io/customer/reactnative/sdk/messagingpush/RNCIOPushMessaging.kt:39]: Thanks for asking Aman. We should expose this in native SDK like we did for `MessagingInApp`. We need to make changes in native SDK first, and can then reuse them here.

[the remark this answers is not in the document]
Comment (ami-aman) [reply android/src/main/java/io/customer/reactnative/sdk/messagingpush/RNCIOPushMessaging.kt:39]: That make sense. I can update these in the Android SDK and open PR to be consumed here.

**Review by mrehan27 — APPROVED** [review]
I see some improvements needed in the Android part, but since this is currently going into feature branch only, let's merge it for now. I’ll make a separate PR for improvements later.

File: android/src/main/java/io/customer/reactnative/sdk/messagingpush/RNCIOPushMessaging.kt (hunk 1 above)
[the remark this answers is not in the document]
Comment (ami-aman) [reply android/src/main/java/io/customer/reactnative/sdk/messagingpush/RNCIOPushMessaging.kt:39]: As per my 1:1 with Ahmed, I will be moving this discussion to the Slack channel to discuss with the team, and then one of us will make the necessary updates.

**Pull request merged**